src/Database/Query.php

```diff
@@ -1412,11 +1413,17 @@ public function values($data)
      *
      * Can be combined with set() and where() methods to create update queries.
      *
-     * @param string $table The table you want to update.
+     * @param string|\Cake\Database\ExpressionInterface $table The table you want to update.
      * @return $this
      */
     public function update($table)
     {
+        if (!is_string($table) && !($table instanceof ExpressionInterface)) {
+            $text = 'Table must be of type string or "%s", got "%s"';
+            $message = sprintf($text, ExpressionInterface::class, gettype($table));
+            throw new InvalidArgumentException($message);
```

Comment:
@thinkingmedia Please check the history of this PR. Everything you mentioned is against what has been developed over time.
The class injected is fine IMO

The gettype() was based on the string of the initial commit. It now turns out there is more to it, and it can be an object as well, but so could also arrays and other types being passed.